diagnostic.md

```diff
@@ -6,72 +6,84 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What is the purpose of a backend?
 
 ```bash
-// your response here
+The back end stores and retrieves the data that your app needs.
 ```
 
 Which layer in the MVC pattern is used by the controller to fetch data?
 
 ```bash
-// your response here
+The model
 ```
 
 Which layer in the MVC pattern communicates with the model?
 
 ```bash
-// your response here
+The controller
 ```
 
 Why don't we use views in our interpretation of the MVC pattern?
 
 ```bash
-// your response here
+Because we are building single page applications
 ```
 
 What does C.R.U.D stand for?
 
 ```bash
-// your response here
+Create Read Update Delete
 ```
 
 In which part of the MVC pattern can we find C.R.U.D actions?
 
 ```bash
-// your response here
+controller
 ```
 
 A user action fires a `GET` request for `person/1`. Explain in detail each step
 required for data to be returned to the client. (bullet points or ordered list)
 
 ```bash
-// your response here
+-The server receives the request and passes it to the appropriate controller. T
+he controller talks to the model
+-the model retrieves the requested data
+-The model passes the data to the controller
+-the controller gives it to the server
+-the server delivers it to the browser/user
 ```
 
 What is the command to generate a new rails-api app?
 
 ```bash
-// your response here
+rails-api new rails-api -T --database=postgresql
 ```
 
 What is the command to start an instance of a rails server?
 
 ```bash
-// your response here
+rails server or rails s
 ```
 
 What are the commands to drop, create and migrate a database? (3 bullet points)
 
 ```bash
-// your response here
+DROP DATABASE
+CREATE DATABASE "databasename";
+\copy databasename(headers) FROM 'path' WITH (FORMAT csv, HEADER true)
+
 ```
 
 What is the command to scaffold a pet with a name and an age?
 
 ```bash
-// your response here
+CREATE TABLE pets (
+  id SERIAL PRIMARY KEY,
+  name TEXT,
+  age INTEGER
+);
 ```
 
 List two advantages of using serializers? (2 bullet points)
 
 ```bash
-// your response here
+Every entry gets a unique id, and you don't need to enter it by hand
 ```
```

Comment:
I'm confused about what you were trying to answer here. You could have wrote something like: hides data and makes it easier to navigate JSON